### PR TITLE
User story 16 - Completed

### PR DESCRIPTION
**Pull request by obellech10**
## Pull Request Review Template

- [X] Wrote Tests
- [X] Implemented
- [] Reviewed

### Necessary checkmarks:
- [X] All Tests are Passing
- [x] The code will run locally

### Type of change
- [X] New feature
- [] Bug Fix

### Implements/Fixes:
Logout functionality 

### Description of Changes:
- Add test for User, Admin, and Merchant User can log out
- Add delete route for logout 
- Add line to log out test to check that flash message appears 
- Change path for nav bar logout link
- Add logout method to user controller
- Update current user to make it a instance variable

closes#
- User Story #16

### Check the correct boxes
- [X] This broke nothing
- [] This broke some stuff
- [] This broke everything

### Testing Changes
- COVERAGE SCORE:
- [X] New Tests have been added
- [] No Tests have been changed
- [] Some Tests have been changed
- [] All of the Tests have been changed (Please describe what in the world happened)

### Checklist:
- [] My code has no unused/commented out code
- [X] I have reviewed my code
- [] I have commented my code, particularly in hard-to-understand areas
- [X] I have fully tested my code

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -20,6 +20,13 @@ def create
   def show
   end
 
+  def logout
+    session.delete(:cart)
```
Comment (smainar) [inline app/controllers/users_controller.rb:29]: I think you can delete lines 24 and 26, and replace it with reset_session. Here's an API doc about it: https://apidock.com/rails/ActionController/Base/reset_session

File: spec/features/users/login_spec.rb [hunk 2]
```diff
@@ -61,5 +61,76 @@
...
+
+        click_button "Login"
+
+        click_link 'Logout'
+
+        expect(current_path).to eq(home_path)
+        expect(page).to have_content("#{admin.name} is now logged out!")
+      end
+    end
+
```
Comment (smainar) [inline spec/features/users/login_spec.rb]: This line can be deleted.

**Comment by smainar** [comment]
@obellech10 - Your testing is thorough. For clarity, my recommendation is to move your tests in a different file: spec/features/users/logout_spec.rb. Currently, your tests are in the login_spec.rb, and this user story pertains to logout functionality.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (obellech10) [reply app/controllers/users_controller.rb:29]: Let's talk about this tomorrow. The reset_session wasn't allowing the flash message to work.

**Comment by obellech10** [comment]
Pulled logout tests out of the login_spec.rb and created new logout_spec.rb file. The reset_session method was messing with the flash message so kept my original code in there for now. Let's review reset_session tomorrow 7/18

**Pull request merged**